docs/expiry-and-auto-renewal/renewal_expiry_design_v2.md

```diff
@@ -0,0 +1,279 @@
+- remove account from `MerkleNetworkContext` new `Queue<AccountNum>`
+
+Pseudo code of execute method:
+
+````java
+class AccountRemovalAction {
+	public execute() {
+		FCMap<MerkleEntityId, MerkleAccount> accounts = getAccounts();
+		accounts.remove(accountNum);
+		merkleNetworkContext.removeAccountForRelRemoval(accountNum);
+	}
+}
+````
+
+The implementation `TokenCleanupAction` will be responsible for:
+
+- iterate over account tokens while checking for available capacity
+- retrieving Merkle token
+- send token from account to treasury
+- remove account to token association
+- delete token from FCMap
+
+Pseudo code of execute method:
+
+````java
+class TokenCleanupAction {
+	public execute() {
+		while (hasEnoughCapacity(available)) {
+			long tokenId = merkleAccount.tokens().getRawIds()[0];
+			MerkleToken token = tokens().get(tokenId);
+			transferToken(accountId, treasury);
+			MerkleEntityAssociation mea = new MerkleEntityAssociation(accountId, tokenId);
+			tokenAssociations().remove(mea);
+          merkleAccount.tokens().remove(0);
+        }
+      FCMap<MerkleEntityId, MerkleAccount> accounts = getAccounts();
+      accounts.remove(accountNum);
+      merkleNetworkContext.removeAccountForRelRemoval(accountNum);
+    }
+}
+````
+
+-------------
+
+### Capacity
+
+The throttling system provides information on the current Hedera transaction load of the network. We can assume what's
+left from the throttle is related to the worker capacity. When the usage of the throttle increases, the worker's
+capacity should decrease accordingly. The capacity of the worker should be split into the buckets. It's important to
+note some buckets should have more capacity than others.
+
+Below is `Capacity` functional interface:
+
+```java
+interface Capacity {
+  long available();
+}
+```
+
+This data represented as long value can be utilized during Hedera transaction and passed to the worker for
+re-calculation. This availability number must be the same for both reads and writes. Different fractions of this number
+will be assigned to all buckets. Each specific action is going to provide a long availability number based on the
+steps/work it needs to process. Once action completes, this number is going to be substracted from the overall worker
+capacity.
+
+As global dynamic properties we'll have two weights:
+
+- touch/read
+- update/write
+
+Update is a manipulation which changes the state, thus it should be more expensive compared to reading.
+
+MAX_CAPACITY integer is always required for the worker. It may be depicted as 100% or just integer MAX_CAPACITY with
+value 100
+
+Options for translating the throttling load to worker capacity are:
+
+1. Reduce MAX_CAPACITY
+2. Increase action's required capacity
+
+Update operations should have higher execution priority. Given the opposite, we will clog the system with operations, as
+the amount of work generated would never be finished in single Hedera transaction.
```

Comment:
I  think that we should iterate on the description here. It is not super clear on how we are going to translate the Throttle load to `capacity`